sale_order_line_variant_description/tests/test_sale_order_line_variant_description.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2015-17 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import TransactionCase
+
+
+class TestSaleOrderLineVariantDescription(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.fiscal_position_model = cls.env["account.fiscal.position"]
+        cls.tax_model = cls.env["account.tax"]
+        cls.pricelist_model = cls.env["product.pricelist"]
+        cls.uom_uom_model = cls.env["uom.uom"]
+        cls.product_tmpl_model = cls.env["product.template"]
+        cls.product_model = cls.env["product.product"]
+        cls.so_model = cls.env["sale.order"]
+        cls.so_line_model = cls.env["sale.order.line"]
+        cls.partner = cls.env.ref("base.res_partner_1")
+
+    def test_product_id_change(self):
+        pricelist = self.pricelist_model.search(
+            [("name", "=", "Public Pricelist")], limit=1
+        )
+        uom = self.uom_uom_model.search([("name", "=", "Units")])[0]
+        tax_include = self.tax_model.create(
+            dict(name="Include tax", amount="0.21", price_include=True)
+        )
+        product_tmpl = self.product_tmpl_model.create(
+            dict(
+                name="Product template",
+                list_price="121",
+                taxes_id=[(6, 0, [tax_include.id])],
+            )
+        )
+        product_tmpl.product_variant_id.update(
+            dict(
+                variant_description_sale="Product variant description",
+            )
+        )
+        product = product_tmpl.product_variant_id
+        fp = self.fiscal_position_model.create(dict(name="fiscal position", sequence=1))
+        so = self.so_model.create(
+            {
+                "partner_id": self.partner.id,
+                "pricelist_id": pricelist.id,
+                "fiscal_position_id": fp.id,
+            }
+        )
+        so_line = self.so_line_model.create(
```

Comment:
In test, the records should be created or written by `Form`  using `from odoo.tests import Form`, so the computes and onchanges will be triggered automatically.